### ISSUE TITLE
IPython.terminal.debugger.set_trace stack contains internals

### ISSUE THREAD

**Issue by amueller** [ipython/ipython#10078]
I think the stack from set_trace should be one level up. Currently you land in the method instantiating the ``TerminalPdb()``:
```python

from IPython.terminal.debugger import set_trace

print("test")
set_trace()
print("bla")
```

I would expect to land on the ``set_trace()`` line. But instead I land in ``debugger.py``:
```
> /home/andy/anaconda3/lib/python3.5/site-packages/IPython/terminal/debugger.py(76)set_trace()
     73             raise
     74 
     75 def set_trace():
---> 76     TerminalPdb().set_trace()
     77 
```

**Comment by minrk** [comment]
Is it possible to set the stack-level? [set_trace](https://docs.python.org/3/library/pdb.html#pdb.set_trace) doesn't take a stack level argument.

**Comment by amueller** [comment]
It's unclear to do that without adding a ``stacklevel`` argument to ``TerminalPdb.set_trace``.

**Comment by minrk** [comment]
I meant the top-level Pdb.set_trace doesn't take a level argument. Is there an easy call to embed pdb at a level above the calling one? If so, we can add a stacklevel arg to TerminalPdb, easily enough.

**Comment by Carreau** [comment]
@amueller can you try on master, that seem to be fixed for me as the line you linked to now says:

```
TerminalPdb().set_trace(frame or sys._getframe().f_back)
```

And that woudl have been part of #9947 which should be backported to 5.2

**Comment by Carreau** [comment]
Closed by #10177 9947
